plugin/src/generators/entity-api/generator.spec.ts

```diff
@@ -0,0 +1,92 @@
+/// <reference types="jest" />
+import * as fs from 'fs';
+import * as path from 'path';
+import { Tree } from '@nx/devkit';
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import * as utils from '../../shared/utils';
+import { entityApiGenerator } from './generator';
+
+jest.mock('enquirer', () => ({
+  AutoComplete: jest.fn().mockImplementation(({ choices }) => ({
+    run: jest.fn().mockResolvedValue(choices[0]),
+  })),
+}));
+
+jest.mock('../../shared/utils', () => ({
+  ...jest.requireActual('../../shared/utils'),
+  getNxLibsPaths: jest.fn(() => ['libs/my-app/shared/data-access/api']),
+  searchNxLibsPaths: jest.fn((paths) => paths.filter((p: string) => p.endsWith('api'))),
+  searchAliasPath: jest.fn(() => '@libs/shared/data-access/api'),
+  appendFileContent: jest.fn(),
+  addNamedImport: jest.fn(),
+}));
+
+jest.mock('ts-morph', () => {
```

Comment:
Let's create a separate file for common mocks. For example

```ts
// jest.setup.js
jest.mock('ts-morph', () => {...});
jest.mock('enquirer', () => {...});

//jest.config.ts
export default {
  ...,
   setupFilesAfterEnv: ['path to jest.setup.js']
}
```

Mocks, actual only for specific generators, can be near their tests